src/app/pages/products-list/popup-host/popup-host.component.ts

```diff
@@ -0,0 +1,45 @@
+import {
+    Component,
+    HostBinding,
+    Input,
+    TemplateRef,
+    ViewChild,
+    ViewContainerRef,
+} from '@angular/core';
+
+@Component({
+    selector: 'app-popup-host',
+    templateUrl: './popup-host.component.html',
+    styleUrls: ['./popup-host.component.css'],
+})
+export class PopupHostComponent {
+    @ViewChild('viewport', {read: ViewContainerRef, static: true})
+    private readonly viewport: ViewContainerRef | undefined;
+
+    @HostBinding('style.display')
+    hostDisplay = 'none';
+
+    @Input() set popupTemplate(template: TemplateRef<unknown> | null) {
+        this.viewport?.clear();
+
+        if (template) {
+            this.hostDisplay = 'block';
+            this.viewport?.createEmbeddedView(template);
+        } else {
+            this.hostDisplay = 'none';
+        }
```

Comment:
Давай вынесем общую часть? Уменьшим дублирование кода

```suggestion
        this.hostDisplay = template ? 'none' : 'block';
            
        if (template) {
            this.viewport?.createEmbeddedView(template);
        }
```